sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-workflowWrapUp.e2e-spec.ts

```diff
@@ -75,4 +97,137 @@ describe("AssessmentController(e2e)-workflowWrapUp", () => {
     expect(expectedAssessment.modifier).toEqual(auditUser);
     expect(expectedAssessment.updatedAt).toBeInstanceOf(Date);
   });
+
+  it("Should find the next impacted assessment and create a reassessment when there is an application for the same student and program year in the future.", async () => {
+    // Arrange
+
+    // Create the student to be shared across the applications.
+    const student = await saveFakeStudent(db.dataSource);
+    // Current application to have the workflow wrapped up.
+    const currentApplicationToWrapUp = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { student },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.Assessment,
+        currentAssessmentInitialValues: {
+          assessmentWorkflowId: "some fake id",
+          assessmentDate: new Date(),
+          studentAssessmentStatus: StudentAssessmentStatus.InProgress,
+        },
+      },
+    );
+    // Resets the workflow data to allow the wrap up worker to be executed.
+    await db.studentAssessment.update(
+      currentApplicationToWrapUp.currentAssessment.id,
+      { workflowData: null },
+    );
+    // Application in the future of the currentApplicationToWrapUp.
+    const impactedApplication = await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { student },
+      {
+        offeringIntensity: OfferingIntensity.partTime,
+        applicationStatus: ApplicationStatus.Completed,
+        currentAssessmentInitialValues: {
+          assessmentWorkflowId: "some fake id",
+          assessmentDate: addDays(1),
+          studentAssessmentStatus: StudentAssessmentStatus.Completed,
+        },
+      },
+    );
+    // Dummy workflowData to be saved during workflow wrap up.
+    const workflowData = {
+      studentData: {
+        dependantStatus: "independant",
+      },
+    } as WorkflowData;
+
+    // Act
+    const result = await assessmentController.workflowWrapUp(
+      createFakeWorkflowWrapUpPayload(
+        currentApplicationToWrapUp.currentAssessment.id,
+        workflowData,
+      ),
+    );
+
+    // Asserts
+    expect(FakeWorkerJobResult.getResultType(result)).toBe(
+      MockedZeebeJobResult.Complete,
+    );
+    // Asserts that the student assessment status has changed to completed.
+    const expectedAssessment = await db.application.findOne({
+      select: {
+        id: true,
+        currentAssessment: {
+          id: true,
+          triggerType: true,
+        },
+      },
+      relations: {
+        currentAssessment: true,
+      },
+      where: {
+        id: impactedApplication.id,
+      },
+    });
+    expect(expectedAssessment.currentAssessment.triggerType).toBe(
+      AssessmentTriggerType.RelatedApplicationChanged,
+    );
+  });
+
+  it("Should return job completed when the workflow data is already set.", async () => {
```

Comment:
I strongly believe that tests like this to ensure idempotency must be there for one per worker which updates data. 👍

Reply:
Yes, I agree too. I believe that we should start a wiki section about worker development recommendations. I will see if I have some time after the PR is done otherwise let's talk about it in the next dev sync-up 😉